bindings/java/src/main/java/org/ucx/jucx/ucp/UcpParams.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2019.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+package org.ucx.jucx.ucp;
+
+import static org.ucx.jucx.ucs.UcsBits.UCS_BIT;
+
+/**
+ * Tuning parameters for UCP library.
+ *
+ * <p>The structure defines the parameters that are used for
+ * UCP library tuning during UCP library ucp_init "initialization".</p>
+ *
+ * <p>UCP library implementation uses the @UcpFeatures "features"
```

Comment:
Need to decide whether to use doxygen or javadoc. Javadoc requires `<p>` tag after the blank line: http://checkstyle.sourceforge.net/config_javadoc.html#JavadocParagraph

Javadoc is more used in java, though doxygen supports java as well.

Reply:
I see, I thought it's doxygen... Does `make docs` build java docs or need to use java tools? is it somehow integrated with "C" UCX docs?

Reply:
I'll add [javadoc plugin](https://maven.apache.org/plugins/maven-javadoc-plugin/usage.html) in separate PR. Then it would be possible to build doc size (mvn site) and build doc as a part of project build.

Reply:
Why not doxygen if it has good java support? the main pros - common index.html, PFD, etc. Also we can transparently link to original "C" docs then it's needed.

Reply:
Ok, i'll change to doxygen. The main concern was that intellij doesn't support dosygen + in google java coding style, that we ported all checks are made for javadoc: https://github.com/openucx/ucx/blob/master/bindings/java/checkstyle.xml#L212

Reply:
I don't push for this, let's ask maintainers before doing this. @yosefe @shamisp @bbenton WDYT?

Reply:
What does doxygen generates when it steps on HTML tag <p> ? It seems like java docs ignores the tags that it does not know. Can we potentially use both ?

If not, I would think that for language bindings we can use the native documentation tool that language defines, since it is natively distributed with the language (in this case JDK). 

BTW, Once we have python bindings ready, we will have similar debate.

@yosefe ?

Reply:
agree, better use javadoc since it probably suits better Java users